components/LoginStatus.tsx

```diff
@@ -0,0 +1,28 @@
+import Image from "next/image";
+import Link from "next/link";
+import { useSession } from "next-auth/react";
+import { signIn, signOut } from "next-auth/react";
+
+export default function LoginStatus() {
+  const { status } = useSession();
+
+  if (status !== "loading") {
```

Comment:
would be easier to read if this was just a separate if statement above `if (status !== "loading")`

then one below that that's `(status === "authenticated")`

the more nested something is, the less readable it is generally

Reply:
think about each state this component can be in and have an if statement for each (except the ending happy path of course)

there are states this component can be in that you aren't handling (loading state for example)

if you don't want it to render, you should explicitly return `null`